packages/dds/sequence/src/test/intervalCollection.fuzz.spec.ts

```diff
@@ -0,0 +1,372 @@
+
+interface IntervalCollectionSpec {
+    collectionName: string;
+}
+
+interface AddText extends ClientSpec {
+    type: "addText",
+    index: number;
+    content: string;
+}
+
+interface RemoveRange extends ClientSpec, RangeSpec {
+    type: "removeRange"
+}
+
+// For non-interval collection fuzzing, annotating text would also be useful.
+
+interface AddInterval extends ClientSpec, IntervalCollectionSpec, RangeSpec {
+    type: "addInterval"
+}
+
+interface ChangeInterval extends ClientSpec, IntervalCollectionSpec, RangeSpec {
+    type: "changeInterval";
+    id: string;
+}
+
+interface DeleteInterval extends ClientSpec, IntervalCollectionSpec {
+    type: "deleteInterval",
+    id: string;
+}
+
+interface Synchronize {
+    type: "synchronize";
+}
+
+type IntervalOperation = AddInterval | ChangeInterval | DeleteInterval;
+
+type TextOperation = AddText | RemoveRange;
+
+type ClientOperation = IntervalOperation | TextOperation;
+
+type Operation = ClientOperation | Synchronize;
+
+// Note: none of these options are currently exercised, since the fuzz test fails with pretty much
+// any configuration due to known bugs. Once shared interval collections are in a better state these
+// should be revisited.
+interface OperationGenerationConfig {
+    /**
+     * Maximum length of the SharedString (locally) before no further AddText operations are generated.
+     * Note due to concurency, during test execution the actual length of the string may exceed this.
+     */
+    maxStringLength?: number;
+    /**
+     * Maximum number of intervals (locally) before no further AddInterval operations are generated.
+     * Note due to concurency, during test execution the actual number of intervals may exceed this.
+     */
+    maxIntervals?: number;
+    maxInsertLength?: number;
+    intervalCollectionNamePool?: string[];
+    validateInterval?: number;
+}
+
+const defaultOptions: Required<OperationGenerationConfig> = {
+    maxStringLength: 1000,
+    maxIntervals: 100,
+    maxInsertLength: 10,
+    intervalCollectionNamePool: ["comments"],
+    validateInterval: 100,
+};
+
+function makeOperationGenerator(optionsParam?: OperationGenerationConfig): Generator<Operation, FuzzTestState> {
+    const options = { ...defaultOptions, ...(optionsParam ?? {}) };
+    type ClientOpState = FuzzTestState & { sharedString: SharedString };
+
+    function isNonEmpty(collection: IntervalCollection<SequenceInterval>): boolean {
+        for (const _ of collection) {
+            return true;
+        }
+
+        return false;
+    }
+
+    // All subsequent helper functions are generators; note that they don't actually apply any operations.
+    function position({ random, sharedString }: ClientOpState): number {
+        return random.integer(0, sharedString.getLength() - 1);
+    }
+
+    function range(state: ClientOpState): RangeSpec {
+        const start = position(state);
+        const end = state.random.integer(start + 1, state.sharedString.getLength());
```

Comment:
should allow start===end, so remove the +1

Reply:
ah, refactored this without thinking hard enough after seeing some errors. We want the current implementation for sharedString modifications (the ranges there are exclusive), but for intervals we should allow start===end. I'll split it up into two methods.